library/swift/src/filters/RequestFilter.swift

```diff
@@ -1,7 +1,28 @@
+@_implementationOnly import EnvoyEngine
 import Foundation
 
+extension EnvoyHTTPFilter {
+  /// Initializer
+  convenience init(requestFilter: RequestFilter) {
```

Comment:
How will this work for filters that want to act as both request and response filters? As-is, it doesn't look like that's possible since the callbacks are only set on `EnvoyHTTPFilter.init`

Reply:
Already by the objective-c mapping layer (and everything below) it's actually unified into a bi-directional solution. It's only bifurcated here because it wasn't clear to me here how we'd work with the two disparate interfaces, and I decided to punt on the problem since it wasn't clear to me that the intent was that we'd have filters that implemented both.

I think I can make things work to allow bi-directional filters with a little tweaking. Since we don't have any usage that necessitates it yet, my slight inclination is to punt on it so we can get something functional onto main and work from there, but I'm open to your thoughts.

Reply:
I don't necessarily want to block the PR on this, but I do think it's required functionality for using these filters in production, and am a little concerned that if we don't fix it early on it'll be more difficult to do as we go further down this route. Thoughts?

Reply:
Just to capture discussion from Friday here, I will be adding support for bi-directionality prior to merging.

Reply:
Pushed a solution, let me know if you have further thoughts.